src/gadgets/nonnative/bn/big_uint_mul_mod_chip/mod.rs

```diff
@@ -150,20 +150,24 @@ impl<F: ff::PrimeField> BigUintMulModChip<F> {
             big_uint::nat_to_f::<F>(&big_uint::get_big_int_with_n_ones(self.limb_width.get()))
                 .unwrap_or_default();
 
-        let (rhs_cells, rhs_tail) = rhs_cells.split_at(self.limbs_count_limit.get());
+        if rhs_cells.len() > self.limbs_count_limit.get() {
```

Comment:
can you explain why previous will panic at `split_at`? Do you mean `rhs_tail.iter()` will panic?

Reply:
Panic [here](https://doc.rust-lang.org/src/core/slice/mod.rs.html#1877)

Len of sum result is less, then limbs limit & panic because of that